cli/cmd/uninstall/uninstall.go

```diff
@@ -0,0 +1,419 @@
+	}
+
+	// Set up the kubernetes client to use for non Helm SDK calls to the Kubernetes API
+	// The Helm SDK will use settings.RESTClientGetter for its calls as well, so this will
+	// use a consistent method to target the right cluster for both Helm SDK and non Helm SDK calls.
+	if c.kubernetes == nil {
+		restConfig, err := settings.RESTClientGetter().ToRESTConfig()
+		if err != nil {
+			c.UI.Output("retrieving Kubernetes auth: %v", err, terminal.WithErrorStyle())
+			return 1
+		}
+		c.kubernetes, err = kubernetes.NewForConfig(restConfig)
+		if err != nil {
+			c.UI.Output("initializing Kubernetes client: %v", err, terminal.WithErrorStyle())
+			return 1
+		}
+	}
+
+	// Setup logger to stream Helm library logs
+	var uiLogger = func(s string, args ...interface{}) {
+		logMsg := fmt.Sprintf(s, args...)
+		c.UI.Output(logMsg, terminal.WithLibraryStyle())
+	}
+
+	c.UI.Output("Existing Installation", terminal.WithHeaderStyle())
+
+	// Search for Consul installation by calling `helm list`. Depends on what's already specified.
+	actionConfig := new(action.Configuration)
+	actionConfig, err := c.initActionConfig(actionConfig, c.flagNamespace, settings, uiLogger)
+	if err != nil {
+		c.UI.Output(err.Error(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	found, foundReleaseName, foundReleaseNamespace, err := c.findExistingInstallation(actionConfig)
+	if err != nil {
+		c.UI.Output(err.Error(), terminal.WithErrorStyle())
+		return 1
+	}
+	if !found {
+		c.UI.Output("No existing Consul installations.", terminal.WithSuccessStyle())
+	} else {
+		c.UI.Output("Existing Consul installation found.", terminal.WithSuccessStyle())
+		c.UI.Output("Consul Un-Install Summary", terminal.WithHeaderStyle())
+		c.UI.Output("Installation name: %s", foundReleaseName, terminal.WithInfoStyle())
+		c.UI.Output("Namespace: %s", foundReleaseNamespace, terminal.WithInfoStyle())
+
+		// Prompt for approval to uninstall Helm release.
+		if !c.flagAutoApprove {
+			confirmation, err := c.UI.Input(&terminal.Input{
+				Prompt: "Proceed with uninstall? (y/N)",
+				Style:  terminal.InfoStyle,
+				Secret: false,
+			})
+			if err != nil {
+				c.UI.Output(err.Error(), terminal.WithErrorStyle())
+				return 1
+			}
+			confirmation = strings.TrimSuffix(confirmation, "\n")
+			if !(strings.ToLower(confirmation) == "y" || strings.ToLower(confirmation) == "yes") {
+				c.UI.Output("Un-install aborted. To learn how to customize the uninstall, run:\nconsul-k8s uninstall --help", terminal.WithInfoStyle())
+				return 1
+			}
+		}
+		// Actually call out to `helm delete`.
+		actionConfig, err = c.initActionConfig(actionConfig, foundReleaseNamespace, settings, uiLogger)
+		if err != nil {
+			c.UI.Output(err.Error(), terminal.WithErrorStyle())
+			return 1
+		}
+		uninstaller := action.NewUninstall(actionConfig)
+		res, err := uninstaller.Run(foundReleaseName)
+		if err != nil {
+			c.UI.Output("unable to uninstall: %s", err, terminal.WithErrorStyle())
```

Comment:
I have a bunch of helper functions throughout that return errors, and then if that function returns an error we print it out to the UI with error style. Since those errors will be lowercased, I was thinking it'd make more sense if any errors we output were lowercased. WDYT?